src/multistream_select/dialer_select.rs

```diff
@@ -346,3 +410,358 @@ impl DialerState {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::multistream_select::listener_select_proto;
+    use std::time::Duration;
+    use tokio::net::{TcpListener, TcpStream};
+
+    #[tokio::test]
+    async fn select_proto_basic() {
+        async fn run(version: Version) {
+            let (client_connection, server_connection) = futures_ringbuf::Endpoint::pair(100, 100);
+
+            let server: tokio::task::JoinHandle<Result<(), ()>> = tokio::spawn(async move {
```

Comment:
nit: Is the type annotation required here and with other `tokio::spawn()` invokations?

Reply:
Unfortunately, I think so because we only return `Ok` from those futures, and the compiler cannot deduce the `Err` type

Reply:
If we never return `Err`, why do we need to return `Result`?

Reply:
Thanks for spotting this! Don't remember why I've added the `Ok()`, probably wanted to use the `?` operator. I've removed the type annotation 🙏